verifier/src/tcb.rs

```diff
@@ -106,21 +106,22 @@ impl TcbInfo {
         Err(Error::UnsupportedTcbLevel)
     }
 
-    fn verify(&self, time: DateTime) -> Result<(), Error> {
+    fn verify(&self, time: Option<DateTime>) -> Result<(), Error> {
         self.verify_version()?.verify_time(time)?;
         Ok(())
     }
 
-    fn verify_time(&self, time: DateTime) -> Result<&Self, Error> {
-        let issue_date = self.issue_date.parse::<DateTime>()?;
-        let next_update = self.next_update.parse::<DateTime>()?;
-        if time < issue_date {
-            Err(Error::TcbInfoNotYetValid)
-        } else if time >= next_update {
-            Err(Error::TcbInfoExpired)
-        } else {
-            Ok(self)
+    fn verify_time(&self, time: Option<DateTime>) -> Result<&Self, Error> {
+        if let Some(date_time) = time {
+            let issue_date = self.issue_date.parse::<DateTime>()?;
+            let next_update = self.next_update.parse::<DateTime>()?;
+            if date_time < issue_date {
+                return Err(Error::TcbInfoNotYetValid);
+            } else if date_time >= next_update {
+                return Err(Error::TcbInfoExpired);
+            }
         }
+        Ok(self)
```

Comment:
IFF the qe_identity logic is updated
```suggestion
        let issue_date = self.issue_date.parse::<DateTime>()?;
        let next_update = self.next_update.parse::<DateTime>()?;
        match time {
            Some(date_time) if date_time < issue_date => Err(Error::TcbInfoNotYetValid),
            Some(date_time) if date_time >= next_update => Err(Error::TcbInfoExpired),
            None | Some(_) => Ok(self),
        }
```